object/map.js

```diff
@@ -0,0 +1,9 @@
+if (!Object.prototype.map) {
+  Object.prototype.map = function(callback) {
+    var array = []
+    Object.keys(this).map((key, index) => {
+       array.push(callback(this[key], key, index))
+    });
+    return array
```

Comment:
you can't use define property for this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty look at the bottom, its just for setting or getting the value on something, if I'm not mistaken. Fixed the other things